### PR TITLE
Wip/user logging

### PR DESCRIPTION
**Pull request by rybmks**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
 import React, { Suspense, lazy } from 'react';
 import './App.css';
 import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
-
-
+import { AppProvider } from './context/AppContext.jsx';
 // Лениво загружаемые компоненты
```
Comment (Hang-Petrov) [inline src/App.js]: Maybe translate that to eng/ua (better eng)?
Or delete it

File: src/pages/shared/navigation/RegisterForm/RegisterFormContent.jsx [hunk 2]
```diff
@@ -18,48 +24,86 @@ function RegistrationForm() {
...
-        const trimmedValues = { ...formValues };
-
-        for (let key in trimmedValues) {
-            if (typeof trimmedValues[key] === 'string') {
-                trimmedValues[key] = trimmedValues[key].trim();
-            }
-        }
-        
+    
+        // Удаление лишних пробелов
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/RegisterForm/RegisterFormContent.jsx]: Same thing with the language. Translate that to eng/ua (better eng) or delete

File: src/pages/shared/navigation/RegisterForm/RegisterFormContent.jsx [hunk 3]
```diff
@@ -18,48 +24,86 @@ function RegistrationForm() {
...
+    
         setValidationError(null);
-
-        if (isRegistering){
-            RegisterUser(trimmedValues);
-        }
-        else {
-            LoginUser(trimmedValues);
+    
+        // Формирование данных для API
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/RegisterForm/RegisterFormContent.jsx]: Same thing with the language

File: src/pages/shared/navigation/RegisterForm/RegisterFormContent.jsx [hunk 4]
```diff
@@ -18,48 +24,86 @@ function RegistrationForm() {
...
         setFormValues(initialValues);
     };
+    
+    const handleApiError = (err) => {
+        if (isRegistering && err.response?.status === 409) {
+            alert("Користувач вже існує!");
+        } else if (!isRegistering && err.response?.status === 404) {
+            alert("Користувача не знайдено! Перевірте логін та пароль!");
+        } else {
+            alert("Сталася помилка. Спробуйте знову пізніше.");
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/RegisterForm/RegisterFormContent.jsx:103]: Maybe without the dot?

File: src/pages/shared/navigation/navigation.jsx [hunk 5]
```diff
@@ -1,56 +1,101 @@
-import "./navigation.css"
+// Стили
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/navigation.jsx]: Language

File: src/pages/shared/navigation/navigation.jsx [hunk 6]
```diff
@@ -1,56 +1,101 @@
...
+import "./navigation.css";
+
+// Хуки
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/navigation.jsx]: Language

File: src/pages/shared/navigation/navigation.jsx [hunk 7]
```diff
@@ -1,56 +1,101 @@
...
 import useClickOutside from "../../../hooks/useClickOutside.js";
 import useWindowWidth from "../../../hooks/useWindowWidth.js";
-import {React, useState, useRef, useEffect } from "react";
+
+// React и хуки
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/navigation.jsx]: Language

File: src/pages/shared/navigation/navigation.jsx [hunk 8]
```diff
@@ -1,56 +1,101 @@
...
+import { React, useState, useRef, useEffect, useContext } from "react";
+
+// Компоненты
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/navigation.jsx]: Language

File: src/pages/shared/navigation/navigation.jsx [hunk 9]
```diff
@@ -1,56 +1,101 @@
...
-import CartIco from  "../icons/Cart_ico.svg"
-import SearchIco from  "../icons/Search_ico.svg"
-import ListIco from  "../icons/TEMP_List_ico.svg"
-import Clear from "../icons/Clear_ico.svg"
 import RegistrationForm from "./RegisterForm/RegisterFormContent.jsx"
 import Modal from "../Modals/Modal.jsx";
-import CartModalContent from "./CartModalContent/CartModalContent.jsx";
+import CartModalContent from './CartModalContent/CartModalContent.jsx';
+
+// Иконки
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/navigation.jsx]: Language

File: src/pages/shared/navigation/navigation.jsx [hunk 10]
```diff
@@ -1,56 +1,101 @@
...
+import UserIco from "../icons/User_ico.svg";
+import CartIco from "../icons/Cart_ico.svg";
+import SearchIco from "../icons/Search_ico.svg";
+import ListIco from "../icons/TEMP_List_ico.svg";
+import Clear from "../icons/Clear_ico.svg";
+
+// API и контексты
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/navigation.jsx]: Language

File: src/pages/shared/navigation/navigation.jsx [hunk 11]
```diff
@@ -1,56 +1,101 @@
...
+import { Requests } from "../../../api/axios_queries/requests.js";
+import axiosClient from "../../../api/axios_queries/axios.js";
+import { AppContext } from "../../../context/AppContext.jsx";
+
+// Роутинг
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/navigation.jsx]: Language

File: src/pages/shared/navigation/navigation.jsx [hunk 12]
```diff
@@ -1,56 +1,101 @@
...
+                if (error.response?.status === 401) {
+                    setIsUserAuthorized(false);
+                    setUserInfo(null);
+                }
+            }
+        }
+            getUserInfo();
+    }, [setIsUserAuthorized, setUserInfo ]);
+
+    // СЛУЖЕБНЫЙ (нужен для вывода юзер даты сразу после того, как она изменилась, потом убрать)
```
Comment (Hang-Petrov) [inline src/pages/shared/navigation/navigation.jsx]: Language - 1
U didn't put an additional info about the time when do we need to delete it
Or I don't see the point of this part there

**Review by Hang-Petrov — CHANGES_REQUESTED** [review]
Small adjustments with the comments, text
Also, I don't see the sense of the commented part or the conditions when do we need to delete it

**Review by FUZIR — APPROVED** [review]

**Review by Hang-Petrov — APPROVED** [review]

**Pull request merged**